crates/bevy_animation/src/lib.rs

```diff
@@ -90,6 +90,58 @@ impl AnimationClip {
     }
 }
 
+/// A representation of the data needed transition to a new animation
+#[derive(Clone, Debug, Reflect)]
+pub struct AnimationTransition {
+    next_animation_clip: Handle<AnimationClip>,
+    repeat: bool,
+    speed: f32,
+    /// The elapsed time of the target clip (is scaled by speed and reset on clip end)
+    clip_elapsed: f32,
```

Comment:
We should be using `Duration` to represent periods of time, not raw f32s, unless we have a very good reason not to.

Reply:
Now using Duration for all fields that represent a period of time.